### PR TITLE
Get recipe button + View updates

### PR DESCRIPTION
**Pull request by lexazhs**
implemented get recipe button in the LoggedInView, updated ResultView so that recipe got from API can display appropriately, updated the way inventory and restriction information is displayed in the InventoryView and RestrictionView, and updated InventoryUseCaseFactory and RestrictionUseCaseFactory since the constructors for InventoryView and RestrictionView changed

### CODE REVIEW

File: src/view/InventoryView.java [hunk 1]
```diff
@@ -128,6 +118,11 @@ public void keyPressed(KeyEvent e) {
             public void keyReleased(KeyEvent e) {
             }
         });
+
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+        this.add(title);
+        this.add(itemInfo);
+        this.add(inventoryInfo);
```
Comment (cher-ning) [inline src/view/InventoryView.java:125]: Consider displaying inventoryInfo before the textbox itemInfo, in terms of UI experience might be better for user to see all their current information first before adding/deleting

Comment (cher-ning) [inline src/view/InventoryView.java:125]: May have missed adding in "button" panel?

File: src/view/RestrictionView.java [hunk 2]
```diff
@@ -118,6 +116,11 @@ public void keyPressed(KeyEvent e) {
             public void keyReleased(KeyEvent e) {
             }
         });
+
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+        this.add(title);
```
Comment (cher-ning) [inline src/view/RestrictionView.java:121]: Does this view also need a textbox (like InventoryView)?

File: src/view/ResultView.java [hunk 3]
```diff
@@ -59,30 +116,17 @@ public void actionPerformed(ActionEvent e) {
...
-
-            @Override
-            public void keyReleased(KeyEvent e) {
-
-            }
-        });
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+        this.add(viewTitle);
+        this.add(usernameInfo);
+        this.add(recipeResult);
```
Comment (cher-ning) [inline src/view/ResultView.java:122]: Remember to add in buttons panel here

**Review by cher-ning — CHANGES_REQUESTED** [review]
Some inconsistencies between InventoryView and RestrictionView and missing elements to address

File: src/view/InventoryView.java (hunk 1 above)
Comment (lexazhs) [reply src/view/InventoryView.java:124]: added, thanks

File: src/view/ResultView.java (hunk 3 above)
Comment (lexazhs) [reply src/view/ResultView.java:121]: added

**Review by cher-ning — APPROVED** [review]

**Pull request merged**